tests/tests.rs

```diff
@@ -432,6 +432,18 @@ sherlock!(context_line_numbers, "world|attached",
     assert_eq!(lines, expected);
 });
 
+sherlock!(max_filesize_parse_error_length, "Sherlock", ".",
+|wd: WorkDir, mut cmd: Command| {
+    cmd.arg("--max-filesize").arg("44444444444444444444");
+    wd.assert_err(&mut cmd);
+});
+
+sherlock!(max_filesize_parse_error_suffix, "Sherlock", ".",
+|wd: WorkDir, mut cmd: Command| {
+    cmd.arg("--max-filesize").arg("45k");
+    wd.assert_err(&mut cmd);
+});
+
```

Comment:
Can you add a test that checks the functionality of `--max-filesize` itself? I realize you have some unit tests, but checking that it works in an integration test will cover some important pieces of code, like the filesize parser and whether the limit is actually passed to the directory iterator.